src/server/game/Achievements/AchievementMgr.cpp

```diff
@@ -65,7 +65,7 @@ bool AchievementMgr::CanUpdateCriteriaTree(Criteria const* criteria, CriteriaTre
 {
     AchievementEntry const* achievement = tree->Achievement;
     if (!achievement)
-        return false;
+        return true;
```

Comment:
What?
If you can't find an achievement linked to the criteriatree, return true no matter what? This is really really bad.